src/model.ts

```diff
@@ -694,7 +694,25 @@ export class SpecQueryModel {
 
         // otherwise, assign the proper to the field def
         if (encQ[prop] !== undefined) {
-          fieldDef[prop] = encQ[prop];
+          if (prop === Property.AXIS) {
```

Comment:
change 

CHANNEL_SUPPORTS_LEGEND[encQ.channel as Channel]

to 

CHANNEL_SUPPORTS_PROPERTY[channel]['legend' / 'scale'/ ...]

and your code will be way shorter